packages/react-paypal-js/src/v6/hooks/useVenmoOneTimePaymentSession.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useCallback, useRef } from "react";
+
+import { usePayPal } from "./usePayPal";
+import { useProxyProps } from "../utils";
+
+import type {
+    VenmoOneTimePaymentSession,
+    VenmoPresentationModeOptions,
+} from "../types";
+import type {
+    UseVenmoOneTimePaymentSessionProps,
+    UseVenmoOneTimePaymentSessionReturn,
+} from "../types/VenmoOneTimePaymentSessionTypes";
+
+export function useVenmoOneTimePaymentSession({
+    presentationMode,
+    createOrder,
+    orderId,
+    ...callbacks
+}: UseVenmoOneTimePaymentSessionProps): UseVenmoOneTimePaymentSessionReturn {
+    const { sdkInstance } = usePayPal();
+    const sessionRef = useRef<VenmoOneTimePaymentSession | null>(null);
+    const proxyCallbacks = useProxyProps(callbacks);
+
+    useEffect(() => {
+        if (!sdkInstance) {
+            throw new Error("no sdk instance available");
+        }
+
+        const newSession = sdkInstance.createVenmoOneTimePaymentSession({
+            orderId,
+            ...proxyCallbacks,
+        });
+        sessionRef.current = newSession;
+
+        return () => {
+            if (sessionRef.current) {
+                sessionRef.current.destroy();
+                sessionRef.current = null;
+            }
+        };
+    }, [sdkInstance, orderId, proxyCallbacks]);
+
+    const handleClick = useCallback(async () => {
```

Comment:
Do we need to implement the click handler for the merchant or is that something they can do? Same for the other handlers. Is the concern just that they may not use `start`, `cancel`, and `destroy` correctly?

If we do go with this pattern, how will we add additional `start` arguments to the top level of arguments for the session hook? I.e. other session types require more `start` options, how should those be passed in? I assume we'd want a similar interface for all of these session hooks if possible.